File: sensorhub-android-app/src/org/sensorhub/android/AppPreferencesActivity.java
package org.sensorhub.android;

import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.content.ServiceConnection;
import android.os.Bundle;
import android.os.IBinder;

import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.appbar.MaterialToolbar;

public class AppPreferencesActivity extends AppCompatActivity {

    SensorHubService boundService;

    private final ServiceConnection sConn = new ServiceConnection() {
        public void onServiceConnected(ComponentName className, IBinder service) {
            boundService = ((SensorHubService.LocalBinder) service).getService();
        }

        public void onServiceDisconnected(ComponentName className) {
            boundService = null;
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.fragment_app_preferences);

        MaterialToolbar toolbar = findViewById(R.id.app_prefs_toolbar);
        setSupportActionBar(toolbar);
        toolbar.setNavigationOnClickListener(v -> onBackPressed());

        bindService(new Intent(this, SensorHubService.class), sConn, Context.BIND_AUTO_CREATE);

        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                .replace(R.id.app_prefs_container, new AppPreferencesFragment())
                .commit();
        }
    }

    @Override
    protected void onDestroy() {
        if (boundService != null) {
            unbindService(sConn);
            boundService = null;
        }
        super.onDestroy();
    }
}

File: sensorhub-android-app/src/org/sensorhub/android/activities/AppPreferencesActivity.kt
package org.sensorhub.android.activities

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import org.sensorhub.android.ui.screens.AppPreferencesScreen
import org.sensorhub.android.ui.theme.OSHTheme

class AppPreferencesActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            OSHTheme {
                AppPreferencesScreen(onBackClick = { finish() })
            }
        }
    }
}